contracts/compliance/modular/modules/TransferFeesModule.sol

```diff
@@ -0,0 +1,210 @@
+ *     T-REX is a suite of smart contracts implementing the ERC-3643 standard and
+ *     developed by Tokeny to manage and transfer financial assets on EVM blockchains
+ *
+ *     Copyright (C) 2023, Tokeny sàrl.
+ *
+ *     This program is free software: you can redistribute it and/or modify
+ *     it under the terms of the GNU General Public License as published by
+ *     the Free Software Foundation, either version 3 of the License, or
+ *     (at your option) any later version.
+ *
+ *     This program is distributed in the hope that it will be useful,
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *     GNU General Public License for more details.
+ *
+ *     You should have received a copy of the GNU General Public License
+ *     along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+pragma solidity 0.8.17;
+
+import "../IModularCompliance.sol";
+import "../../../token/IToken.sol";
+import "../../../roles/AgentRole.sol";
+import "./AbstractModule.sol";
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+contract TransferFeesModule is AbstractModule, Ownable {
+    /// Struct of fees
+    struct Fee {
+        uint256 rate; // min = 0, max = 10000, 0.01% = 1, 1% = 100, 100% = 10000
+        address collector;
+    }
+
+    /// Mapping for compliance fees
+    mapping(address => Fee) private _fees;
+
+    /**
+    *  this event is emitted whenever a fee definition is updated for the given compliance address
+    *  the event is emitted by 'setFee'.
+    *  compliance is the compliance contract address
+    *  _rate is the rate of the fee (0.01% = 1, 1% = 100, 100% = 10000)
+    *  _collector is the collector wallet address
+    */
+    event FeeUpdated(address indexed compliance, uint256 _rate, address _collector);
+
+    error FeeRateIsOutOfRange(address compliance, uint256 rate);
+
+    error InvalidFeeCollectorAddress(address compliance, address collector);
+
+    error CollectorAddressIsNotVerified(address compliance, address collector);
+
+    /**
+    *  @dev Sets the fee rate and collector of the given compliance
+    *  @param _compliance is the compliance contract address
+    *  @param _rate is the rate of the fee (0.01% = 1, 1% = 100, 100% = 10000)
+    *  @param _collector is the collector wallet address
+    *  Only the owner of the token smart contract can call this function
+    *  Collector wallet address must be verified
+    */
+    function setFee(address _compliance, uint256 _rate, address _collector) external {
+        address tokenAddress = IModularCompliance(_compliance).getTokenBound();
+        require(msg.sender == AgentRole(tokenAddress).owner(), "only token owner can call");
+
+        if (_rate > 10000) {
+            revert FeeRateIsOutOfRange(_compliance, _rate);
+        }
+
+        if (_collector == address(0)) {
```

Comment:
if the address is zero address `isVerified` that is called right after will return `false` therefore we can remove this check here